### DISCUSSION TITLE
Installation: Invalid MEX-file

### DISCUSSION THREAD

**Discussion by grld** [mtex-toolbox/mtex#2003] [Ask Anything] [answered]
Dear community,

I tried to install MTEX on our university cluster. After trying first time to directly do startup_mtex, I received error messages that NFFT libraries couldn't get run. So I followed the instructions and compiled NFFT from scratch. Everything worked, I did the `./configure` with `--enable-all` and all the fftw flag, and matlab flag described in the MTEX readme [here](https://github.com/mtex-toolbox/mtex/blob/develop/extern/nfft_openMP/readme.md). 
Make install worked without any issues, only lib giving me that info:

> Libraries have been installed in:
> /storage/home/hcoda1/7/jschmidt87/software/nfft-3.5.3-installed/lib
> 
> If you ever happen to want to link against installed libraries
> in a given directory, LIBDIR, you must either use libtool, and
> specify the full pathname of the library, or use the `-LLIBDIR'
> flag during linking and do at least one of the following:
> - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
> during execution
> - add LIBDIR to the `LD_RUN_PATH' environment variable
> during linking
> - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
> - have your system administrator add LIBDIR to `/etc/ld.so.conf'
> See any operating system documentation about shared libraries for
> more information, such as the ld(1) and ld.so(8) manual pages.
> ----------------------------------------------------------------------

Then I copied the executables `fptmex.mexa64, nfftmex.mexa64, nfsftmex.mexa64, nfsoftmex.mexa64` to the mtex dir `mtex-5.10.2/extern/nfft_openMP/` as well as to its subdir `/mtex-5.10.2/extern/nfft_openMP/mtex-5.7.0`.

So I followed all the steps I could find in the readme, but still face the following error when doing startup_mtex:
`MTEX: Error running NFFT with openMP - I'm going to try it without openmp 
[...]
  The original error message was
Invalid MEX-file
"/storage/home/hcoda1/7/jschmidt87/software/mtex-5.10.2/mex/adjoint_representationbased_coefficient_transform.mexa64":
/lib64/libm.so.6: version "GLIBC_2.29" not found (required by
/storage/home/hcoda1/7/jschmidt87/software/mtex-5.10.2/mex/adjoint_representationbased_coefficient_transform.mexa64)
[...]`

Any ideas why MTEX can not use my compiled binaries? Appreciate it!

**Comment by kilir** [comment, marked as answer]
Hi,
so for now just the non-openMP binaries work?
I assume you compiled the .mex files on the machine on which you are going to use them and you have openMP installed. Did you try the generic ones here: https://www-user.tu-chemnitz.de/~potts/nfft/download.php#matlab

The second error with "adjoint_representationbased_coefficient_transform" says that your glibc is < 2.29. Since glibc comes with ldd you could check the version with `ldd --version`. You can try to build those .mex files running mex/mex_install.m.

Hope this helps.

Cheers,
Rüdiger

**Reply by grld** [reply]
Hi Rüdiger,

I haven't tried to recompile the non-openMP ones. I shortened the error message above. MTEX also tries them after not getting the ones with openMP to work, and fails for the non-openMP ones as well. I will try to use the generic ones.

I checked my GNU C lib version and it's 2.17 :D So maybe a little outdated. I used your hint and did `mex_install(PATH_TO_MTEX)`. It compiled MEX successfully. Then I tried `startup_mtex` and it worked smoothly now :) Big thanks!!

Kind regards,
Jan
